Source/monster.cpp

```diff
@@ -3891,8 +3886,16 @@ void M_StartHit(Monster &monster, const Player &player, int dam)
 
 void MonsterDeath(Monster &monster, Direction md, bool sendmsg)
 {
+	int monsterExp = sgGameInitInfo.nDifficulty == DIFF_NIGHTMARE ? 2 * (monster.exp + 1000) : sgGameInitInfo.nDifficulty == DIFF_HELL ? 4 * (monster.exp + 1000)
+	                                                                                                                                   : monster.exp;
+	if (monster.isUnique()) {
+		monsterExp *= 2;

```

Comment:
Should this bug be fixed? Probably wasn't intended to run the difficulty adjustment again for unique monsters 🤷🏽‍♂️

Reply:
+ you should leave old values for vanilla compatibility or monsters will give 0 xp when you load devx save in vanilla xD

Reply:
@qndel is this a response to my other comment? If so, that's why I was saying the old values (from the array) should be saved to the save file, but we don't need to reload them in DevX.